HVZ.Web/Shared/Org Settings/OrgGeneralSettings.razor

```diff
@@ -0,0 +1,201 @@
+    <button class="btn btn-danger" @onclick="Discard">Discard</button>
+</div>
+
+@code
+{
+    [CascadingParameter(Name = "OrgId")]
+    public string? OrgId { get; set; }
+
+    Organization? org;
+    ImageUploader? uploader;
+    IDisposable navigationHandler = null!;
+
+    string orgName = "";
+    string orgDesc = "";
+    bool stagedFile;
+    bool requireVerifiedEmail = false;
+    bool requireProfilePicture = false;
+
+    string targetLocation = "";
+
+    bool UnsavedChanges =>
+        (orgName != org?.Name) ||
+        // Org Description
+        (orgDesc != org?.Description) ||
+        // Verified email
+        (requireVerifiedEmail != org?.RequireVerifiedEmailForPlayer) ||
+        // Require profile picture
+        (requireProfilePicture != org?.RequireProfilePictureForPlayer) ||
+        stagedFile;
+
+    string SaveFooterVisibility => UnsavedChanges ? "d-block" : "d-none";
+
+    protected override bool ShouldRender() => OrgId != null;
+
+    protected override async Task OnInitializedAsync()
+    {
+        navigationHandler = NavigationManager.RegisterLocationChangingHandler(LocationChangeHandler);
+        await base.OnInitializedAsync();
+    }
+
+    void GetOrgParameters()
+    {
+        if (org is null) return; // Does not matter this was already handled
+        orgName = org.Name;
+        orgDesc = org.Description;
+        requireProfilePicture = org.RequireProfilePictureForPlayer;
+        requireVerifiedEmail = org.RequireVerifiedEmailForPlayer;
+    }
+
+    private ValueTask LocationChangeHandler(LocationChangingContext context)
+    {
+
+        if (UnsavedChanges)
+        {
+            Console.WriteLine(UnsavedChanges);
+            targetLocation = context.TargetLocation;
+            context.PreventNavigation();
+            JS.InvokeVoidAsync("showModalById", "saveChangesModal");
+        }
+
+        return ValueTask.CompletedTask;
+    }
+
+    protected override async Task OnParametersSetAsync()
+    {
+        if (OrgId is null) return;
+
+        org = await OrgRepo.FindOrgById(OrgId);
+        if (org is null) throw new ArgumentNullException("How the fuck did you even get here?");
+        GetOrgParameters();
+    }
+
+    async Task SaveAndContinue()
+    {
+        await Save();
+        NavigationManager.NavigateTo(targetLocation, true);
+    }
+
+    void DiscardAndContinue()
+    {
+        Discard();
+        NavigationManager.NavigateTo(targetLocation, true);
+    }
+
+    async Task Save()
+    {
+        if (uploader is not null)
+        {
+            await uploader.SaveFile();
+            uploader.ClearFile();
+        }
+        if (OrgId is null) throw new ArgumentException("Seriously how the fuck did you even get here what the actual fuck is going on");
+        await OrgRepo.SetOrgDescription(OrgId, orgDesc);
+        await OrgRepo.SetRequireProfilePicture(OrgId, requireProfilePicture);
+        await OrgRepo.SetRequireVerifiedEmail(OrgId, requireVerifiedEmail);
```

Comment:
TODO: create bulk settings update method to simplify the database calls required in situations like this